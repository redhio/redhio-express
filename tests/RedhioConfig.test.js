const redhioExpress = require('../index');

describe('RedhioConfig', async () => {
  const originalConsoleError = console.error;
  beforeEach(() => {
    console.error = jest.fn();
  });

  afterEach(() => {
    console.error = originalConsoleError;
  });

  it('logs errors when given empty object', () => {
    redhioExpress({});
    expect(console.error).toBeCalled();
    expect(console.error.mock.calls).toMatchSnapshot();
  });

  it('logs errors when given bad props', () => {
    redhioExpress({
      apiKey: 32,
      host: { notGood: true },
      secret: true,
      scope: 'orders',
      afterAuth: true,
      accessMode: 'gerblable',
    });
    expect(console.error).toBeCalled();
    expect(console.error.mock.calls).toMatchSnapshot();
  });

  it('does not log errors when given valid proptypes', () => {
    redhioExpress({
      apiKey: 'fake',
      host: 'fake',
      secret: 'cats',
      scope: [],
      afterAuth: () => null,
    });
    expect(console.error).not.toBeCalled();
  });
});
