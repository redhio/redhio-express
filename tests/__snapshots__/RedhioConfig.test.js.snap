// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopifyConfig logs errors when given bad props 1`] = `
Array [
  Array [
    "Warning: Failed option type: Invalid option \`apiKey\` of type \`number\` supplied to \`RedhioExpress\`, expected \`string\`.",
  ],
  Array [
    "Warning: Failed option type: Invalid option \`host\` of type \`object\` supplied to \`RedhioExpress\`, expected \`string\`.",
  ],
  Array [
    "Warning: Failed option type: Invalid option \`secret\` of type \`boolean\` supplied to \`RedhioExpress\`, expected \`string\`.",
  ],
  Array [
    "Warning: Failed option type: Invalid option \`scope\` of type \`string\` supplied to \`RedhioExpress\`, expected an array.",
  ],
  Array [
    "Warning: Failed option type: Invalid option \`afterAuth\` of type \`boolean\` supplied to \`RedhioExpress\`, expected \`function\`.",
  ],
  Array [
    "Warning: Failed option type: Invalid option \`accessMode\` of value \`gerblable\` supplied to \`RedhioExpress\`, expected one of [\\"offline\\",\\"online\\"].",
  ],
]
`;

exports[`ShopifyConfig logs errors when given empty object 1`] = `
Array [
  Array [
    "Warning: Failed option type: The option \`apiKey\` is marked as required in \`RedhioExpress\`, but its value is \`undefined\`.",
  ],
  Array [
    "Warning: Failed option type: The option \`host\` is marked as required in \`RedhioExpress\`, but its value is \`undefined\`.",
  ],
  Array [
    "Warning: Failed option type: The option \`secret\` is marked as required in \`RedhioExpress\`, but its value is \`undefined\`.",
  ],
  Array [
    "Warning: Failed option type: The option \`scope\` is marked as required in \`RedhioExpress\`, but its value is \`undefined\`.",
  ],
  Array [
    "Warning: Failed option type: The option \`afterAuth\` is marked as required in \`RedhioExpress\`, but its value is \`undefined\`.",
  ],
]
`;
